Source/Core/Core/PowerPC/PPCTables.h

```diff
@@ -89,32 +88,26 @@ struct GekkoOPInfo
 {
   const char* opname;
   OpType type;
-  int flags;
-  int numCycles;
-  u64 runCount;
-  int compileCount;
-  u32 lastUse;
+  u32 flags;
 };
-extern std::array<GekkoOPInfo*, 64> m_infoTable;
-extern std::array<GekkoOPInfo*, 1024> m_infoTable4;
-extern std::array<GekkoOPInfo*, 1024> m_infoTable19;
-extern std::array<GekkoOPInfo*, 1024> m_infoTable31;
-extern std::array<GekkoOPInfo*, 32> m_infoTable59;
-extern std::array<GekkoOPInfo*, 1024> m_infoTable63;
 
-extern std::array<GekkoOPInfo*, 512> m_allInstructions;
-extern size_t m_numInstructions;
+// The current (half-open) ranges of OpID.
+// An OpID identifies a distinctive instruction encoding and constitutes
+// a shared concept of a 'PPC operation' among the JITs and other
+// CPU emulation components.
+// Use PPCTables::GetOpId to get the right one for a given instruction.
+enum class OpId : int
+{
+  Invalid = 0,
+#include "OpID_Ranges.gen.h"
```

Comment:
This should be using a full relative path like how all our includes do.

Reply:
As before, I'm open to all suggestions on how to integrate code generation with the build better.